lib/spack/docs/build_systems/gopackage.rst

```diff
@@ -0,0 +1,164 @@
+
+        depends_on('go@1.13:', type='build')  # go.mod value overrides default
+
+        executables = ['lazygit']
+
+While it generally follows the common patterns of a Spack package,
+it's worth calling attention to its:
+
+* adding a *build* dependency on the version of the Go package that it
+  requires (from the existing `go.mod`);
+* using the default ``build`` and ``install`` steps that ``GoPackage``
+  provides; and
+* providing the name of the executable to be installed by the default
+  ``install`` step.
+
+The ``GoPackage`` provides a `setup_build_environment` that sets
+`GO111MODULES=on` and `GOFLAGS=-mod=vendor` and its default build step
+invokes ``go build``.  Additional arguments to ``go build`` can be
+provided by setting ``build_args``, e.g. one could
+
+.. code-block:: python
+
+        build_args = ['-p', '12']  # adjust parallelism of build cmd
+
+to invoke ``go build -p 12``.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Slightly more complicated packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Packages that can be built with a simple invocation of ``go build``,
+that use modules, *but have not vendored their dependencies* require a
+bit of extra work.  In particular, in order to satisfy Spack's
+requirement that nothing is downloaded during the build step, the
+package needs to define "resources" that describe from whence to fetch
+each dependency, where to emplace it, and the version of the package
+to which it applies.  It is possible (likely) that each release of the
+project will new set of resources definitions covering the correct
+versions.
+
+In the general case, compiling these resource is complicated.
+
+Fortunately, packagers of projects that use modules can use an
+automated tool (``modules2tuple``) to provide the resource definitions
+and ``import_resources`` to include them without overly-complicating
+the package definition.  ``modules2tuple``
+(https://github.com/dmgk/modules2tuple) was originally developed by
+the FreeBSD Ports team to generate the Makefile fragment their build
+system uses to represent a Go application's dependencies.  For the
+curious, their Porters Handbook includes more details about `packaging
+Go Applications
+<https://www.freebsd.org/doc/en_US.ISO8859-1/books/porters-handbook/building.html#using-go>`_
+for FreeBSD.
+
+There are three steps to describing the dependencies in the package
+definition:
+
+* First, stage the application and use the Go tool chain to
+  solve the set of required dependencies and build a vendor directory.
+
+* Second, run the newly created ``vendor/modules.txt`` file through
+  ``module2tuples`` to generate a JSON file containing resource
+  definitions.
+
+.. code-block:: console
+
+    # in the root directory of the project (where the go.mod file lives)
+    $ go mod vendor
+    $ modules2tuple -spack -app_version=1.2.3 vendor/modules.txt > resources-1.2.3.json
+
+* Finally, copy that file into the Spack package directory and add an
```

Comment:
I would use `#.` instead of `*` to make a numbered list for these since they're clearly ordered.

Reply:
Every day, I learn something new....